src/main/java/com/thealgorithms/scheduling/JobSchedulingWithDeadline.java

```diff
@@ -0,0 +1,83 @@
+package com.thealgorithms.scheduling;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * A class that implements a job scheduling algorithm to maximize profit
+ * while adhering to job deadlines.
+ *
+ * This class provides functionality to schedule jobs based on their profit
+ * and deadlines to ensure that the maximum number of jobs is completed
+ * within the given timeframe. It sorts the jobs in decreasing order of profit
+ * and attempts to assign them to the latest possible time slots.
+ */
+public final class JobSchedulingWithDeadline {
+    private JobSchedulingWithDeadline() {
+    }
+
+    /**
+     * Represents a job with an ID, deadline, and profit.
+     *
+     * Each job has a unique identifier, a deadline by which it must be completed,
+     * and a profit associated with completing the job.
+     */
+    static class Job {
+        int jobId;
+        int deadline;
+        int profit;
```

Comment:
Shouldn't job arrival time be included as well? It's present in other scheduling algorithms

Reply:
Included arrival time in the algorithm.